RELEASE/scripts/autoscend.ash

```diff
@@ -5258,42 +5254,67 @@ boolean LX_loggingHatchet()
 
 boolean LX_getStarKey()
 {
-	if (internalQuestStatus("questL10Garbage") < 9 || internalQuestStatus("questL11Shen") < 7)
+	//if we don't want the star key then skip this function
+	if(!get_property("auto_getStarKey").to_boolean())
 	{
 		return false;
 	}
-	if(!get_property("auto_getStarKey").to_boolean())
+
+	//if already have the star key or already used it, then stop running this function.
+	if(!needStarKey())
+	{
+		set_property("auto_getStarKey", false);
+		return false;
+	}
+	
+	//if we did not progress enough in giant garbage quest to reach hole in the sky then skip this function
+	//if shen quest is not far ahead enough also skip. because shen might send us there and it would be a waste of adventures if we already got the star key before he sends us there.
+	if (internalQuestStatus("questL10Garbage") < 9 || internalQuestStatus("questL11Shen") < 7)
 	{
 		return false;
 	}
+	
+	//can we reach the hole in the sky. need rocketship, except in kingdom of exploathing.
 	if(item_amount($item[Steam-Powered Model Rocketship]) == 0 && !in_koe())
 	{
 		return false;
 	}
-	if(!needStarKey())
+	
+	//pull a star chart when all following are true: not hardcore. at the door. no chart. no key. key not used.
+	if (!in_hardcore() && internalQuestStatus("questL13Final") == 5 && item_amount($item[Richard\'s Star Key]) == 0 && item_amount($item[Star Chart]) == 0 && !get_property("nsTowerDoorKeysUsed").contains_text("Richard's star key"))
+	{
+		pullXWhenHaveY($item[Star Chart], 1, 0);
+	}
+	
+	//craft the star key if I have the ingredients for it
```

Comment:
I think you have a general habit of adding comments when it's super obvious what the underlying code does. Comments are really useful for when code is doing something atypical, fancy, or is making assumptions that aren't obvious. But not all code has to be accompanied by a comment, especially if it's as simple as "if we don't have any X and can craft X, craft X".

The risk of having too many comments is it obscures the portions of the codebase that are *genuinely* complicated and need comments. Fun anecdote: at work, I once spent two hours reviewing a 1000-line change that contained 500 lines of comments. It was very scary code and it 100% needed those comments, but if every change was 50% comments than it wouldn't be obvious that that particular section of the code was scary.